lib/openstax/rescue_from/logger.rb

```diff
@@ -0,0 +1,31 @@
+module OpenStax
+  module RescueFrom
+    class Logger
+      attr_reader :wrapped
+
+      def initialize(wrapped)
+        @wrapped = wrapped
+      end
+
+      def record_system_error!
```

Comment:
The `header` is only used here as the first part of the log message, and I think what you're trying to achieve is to have "An exception occurred" for the top-level guy followed by "Exception cause" for all of the recursive calls to log the causes -- is that right?

If that's the case, I think you should move `header` out of the wrapped exception and just have it be a `Logger` thing only, e.g.

```ruby
def record_system_error!(prefix = "An exception occurred")
  config.logger.error("#{prefix}: ...")
  record_system_error_recursively!
end

def record_system_error_recursively!
  ...
  record_system_error!("Exception cause") 
end
```